app/src/main/java/com/reminder/hydration/SharedPrefManager.kt

```diff
@@ -0,0 +1,52 @@
+package com.reminder.hydration
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class SharedPrefManager private constructor(context: Context) {
+    private val sharedPreferences: SharedPreferences =
+        context.getSharedPreferences("Hydration", Context.MODE_PRIVATE)
+
+
+    fun <T> save(key: String, value: T) {
+        when (value) {
+            is String -> {
+                sharedPreferences.edit()
+                    .putString(key, value)
+                    .apply()
```

Comment:
you are doing this call 4 times. Use the variable you created above at the bottom of the function and call this function.